src/sentry/replays/usecases/query/__init__.py

```diff
@@ -223,8 +224,22 @@ def query_using_optimized_search(
             SearchFilter(SearchKey("environment"), "IN", SearchValue(environments)),
         ]
 
-    # Translate "viewed_by_me" filters, which are aliases for "viewed_by_id"
-    search_filters = handle_viewed_by_me_filters(search_filters, request_user_id)
+    viewed_by_denylist = options.get("replay.viewed-by.project-denylist")
+    if any([project_id in viewed_by_denylist for project_id in project_ids]):
+        # Skip all viewed by filters if in denylist
+        new_filters = []
+        for search_filter in search_filters:
+            if search_filter.key.name not in [
+                "viewed_by_id",
+                "viewed_by_me",
+                "seen_by_id",
+                "seen_by_me",
+            ]:
+                new_filters.append(search_filter)
+        search_filters = new_filters
```

Comment:
This could have been simplified to:

```python
filter(lambda x: x.key.name not in [...], search_filters)
```

But! I think we should error. We don't want to silently fail and have people thing the app is broken.  Let's 400 and return an error message.

Reply:
Also the viewed by fields need to be a constant.  Similar to https://github.com/getsentry/sentry/blob/997cf1157084c4186b7714c0a51f6a2dfc86b969/src/sentry/replays/usecases/query/__init__.py#L48